src/Extensions/Hosting.Services.Web/Middlewares/UserIdentiyMiddleware.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.Net;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Omex.Extensions.Abstractions.Activities;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web.Middlewares
+{
+	/// <summary>
+	/// Adds user information if needed
+	/// </summary>
+	internal class UserIdentiyMiddleware : IMiddleware, IDisposable
```

Comment:
Is the intention that we extend this internally to use identity information and that the public version only uses IPs?

Reply:
My initial intension was that everybody would be using just IP, but I've abstracted it under `IUserIdentityProvider` so implementations could be replaced with some other identity if needed

Reply:
Good, an IP only identifies a device, not a user.
If a user is authenticated their user ID is a better identifier, since it follows them across devices.